utils/prettier.js

```diff
@@ -0,0 +1,184 @@
+/* global prettier, prettierPlugins */
+import PrettierIcon from '../components/PrettierIcon.js';
+import CheckMarkIcon from '../components/PrettierCheckMarkIcon.js';
+import LoadingIcon from '../components/PrettierLoadingIcon.js';
+import ErrorIcon from '../components/PrettierErrorIcon.js';
+
+/**
+ * UTILS
+ */
+
+const getFinalExtension = ext =>
+  ext ? ext.slice(ext.lastIndexOf('.') + 1) : ''; // Also works when no `.` present in string
+
+/**
+ * STATE SETUP
+ */
+
+const prettierButtonMessages = {
+  before: 'Format Code',
+  during: 'Formatting...',
+  after: 'Formatted',
+  error: 'Formatting Failed',
+};
+
+const beforePrettierState = {
+  message: prettierButtonMessages.before,
+  disabled: false, // redundant?
+  icon: PrettierIcon,
+};
+
+const duringPrettierState = {
+  message: prettierButtonMessages.during,
+  disabled: true,
+  icon: LoadingIcon,
+};
+
+const afterPrettierState = {
+  message: prettierButtonMessages.after,
+  disabled: true,
+  icon: CheckMarkIcon,
+};
+
+const errorPrettierState = {
+  message: prettierButtonMessages.error,
+  disabled: true,
+  icon: ErrorIcon,
+};
+
+const cannotPrettierState = {
+  hidden: true,
+  icon: null,
+};
```

Comment:
🆒 way to handle the UI states

Reply:
🤗 Yeah I prefer the neatness of having these separated out clearly! Probably influenced by TS enums and Redux